tasks.star

```diff
@@ -358,22 +359,93 @@ def configure():
         ],
     )
 
+    db_network = "nebula"
+    db_container = "nebula-db"
+    db_port = 1234
+    db_user = "nebula"
+    db_pass = "nebula"
+    db_name = "nebula"
+    db_url = "postgres://%s:%s@localhost:%d/%s" % (db_user, db_pass, db_port, db_name)
+
+    task(
+        "database-setup",
+        hidden = True,
+        skip_if_exists = [
+            ".tools/db_setup"
+        ],
+        cmds = [
+            "docker network create '%s'" % db_network,
+            "docker create --name '%s' --network '%s' -p '%d:5432' -e POSTGRES_USER='%s' -e POSTGRES_PASSWORD='%s' -e POSTGRES_DB='%s' postgres:alpine" % (db_container, db_network, db_port, db_user, db_pass, db_name),
+            "touch .tools/db_setup"
+        ]
+    )
+
+    task(
+        "database-ready",
+        hidden = True,
+        deps = ["database-setup"],
+        cmds = [
+            "docker start '%s'" % db_container,
+            "until docker exec '%s' pg_isready; do sleep 1; done" % db_container,
+        ]
+    )
+
+    task(
+        "database-migrate",
+        desc = "Initializes and migrates the Nebula database (using Docker)",
+        deps = ["database-ready"],
+        inputs = [
+            "db/migrations/*.sql"
+        ],
+        outputs = [
+            ".tools/db_migrated"
+        ],
+        cmds = [
+            "docker run --rm --network '%s' -v \"$PWD/db/migrations:/flyway/sql\" flyway/flyway:latest-alpine -url='jdbc:postgresql://%s/%s?user=%s&password=%s' migrate" % (db_network, db_container, db_name, db_user, db_pass),
+            "touch .tools/db_migrated"
+        ]
+    )
+
+    task(
+        "database-clean",
+        desc = "Tears down the managed Nebula database",
+        deps = ["build-tool"],
+        ignore_exit = True,
+        cmds = [
+            "docker rm -f '%s'" % db_container,
+            "docker network rm '%s'" % db_network,
+            "rm .tools/db_*"
+        ]
+    )
+
+    neb_bin = resolve_path("./build/nebula%s" % binext)
+
     task(
         "server-build",
         desc = "Compiles the Nebula server code",
-        deps = ["proto-build"],
+        deps = ["proto-build", "database-migrate"],
         inputs = [
             "packages/server/cmd/**/*.go",
             "packages/server/pkg/**/*.go",
         ],
         outputs = ["build/nebula%s" % binext],
+        base = "packages/server",
```

Comment:
Since you added `base`, you'll have to remove the `packages/server` prefixes here as well. I usually make `base` the first parameter after `deps` to make this more obvious.

Reply:
Aha! Spotted an oversight too.